### PR TITLE
ARROW-1236: Fix lib path in pkg-config file

### PR DESCRIPTION
**Pull request by Zaharid**
The `CMAKE_INSTALL_LIBDIR` already contains the prefix on a default
installation (such as in the conda-forge package) and therefore
prepending the prefix results in the wrong path.

While apparently it is not clear what this variable should be
(https://bugzilla.redhat.com/show_bug.cgi?id=795542), it seems that
it's used as an absolute path in the rest of the project.

### CODE REVIEW

**Comment by wesm** [comment]
@kou could you take a look?

@Zaharid would you mind creating a JIRA for the changelog and adding it to the PR title? Thank you!

File: cpp/src/arrow/arrow.pc.in [hunk 1]
```diff
@@ -16,7 +16,7 @@
 # under the License.
 
 prefix=@CMAKE_INSTALL_PREFIX@
-libdir=${prefix}/@CMAKE_INSTALL_LIBDIR@
+libdir=@CMAKE_INSTALL_LIBDIR@
 includedir=${prefix}/include
```
Comment (kou) [inline cpp/src/arrow/arrow.pc.in]: How about this?

```diff
-prefix=@CMAKE_INSTALL_PREFIX@
-libdir=${prefix}/@CMAKE_INSTALL_LIBDIR@
-includedir=${prefix}/include
+libdir=@CMAKE_INSTALL_FULL_LIBDIR@
+includedir=@CMAKE_INSTALL_FULL_INCLUDEDIR@
```

**Comment by kou** [comment]
Please add ":" after "ARROW-1236" like "ARROW-1236: Fix ...".

**Comment by Zaharid** [comment]
Cool, seems this worked. The JDK test failures look unrelated.

**Comment by kou** [comment]
+1

**Comment by kou** [comment]
Thanks!

**Pull request closed without merging**